llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -921,6 +950,46 @@ static std::string getESIMDIntrinSuffix(id::FunctionEncoding *FE,
   return Suff;
 }
 
+static void translateBlockLoad(CallInst &CI, bool IsSLM) {
+  IRBuilder<> Builder(&CI);
+
+  APInt Val =
+      parseTemplateArg(CI, 2, ESIMDIntrinDesc::GenXArgConversion::TO_I64);
+  MaybeAlign Align(Val.getZExtValue());
+
+  auto Op0 = CI.getArgOperand(0);
+  auto DataType = CI.getType();
+  if (IsSLM) {
```

Comment:
can you test with opaque pointers and make sure this inttoptr conversion works? maybe the types are different

Reply:
Yes, I did the tests.
Also, added few more tests (1 check to existing LIT test) and added 1 E2E test in 1e722d1af21d4d979f9e352e053a0ae4d340e45a

Thank you for this comment. The additional testing showed some misses in the initial version of the patch.